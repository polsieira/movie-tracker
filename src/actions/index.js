import { postFavorite, deleteFavorite } from '../apiCalls'

export const addMovies = movies => {
  return ({
    type: 'ADD_MOVIES',
    movies
  })
}

export const hasErrored = message => {
  return ({
    type: 'HAS_ERRORED',
    message
  })
}

export const checkIsLoading = bool => {
  return ({
    type: 'CHECK_IS_LOADING',
    bool
  })
}

export const loginUser = ({ name, id, isSignedIn }) => {
  return ({
    type: 'LOGIN_USER',
    name,
    id,
    isSignedIn
  })
}

export const createUser = ({ name, email, password }) => {
  return ({
    type: 'CREATE_USER',
    name,
    email,
    password
  })
}

export const getFavorites = faves => {
  return ({
    type: 'GET_FAVORITES',
    faves
  })
}

export const addFavorite = favorite => {
  return ({
    type: 'ADD_FAVORITE',
    favorite
  })
}

export const fetchAndPostFavorite = (id, favorite) => {
  return (dispatch) => {
    return postFavorite(id, favorite)
      .then(result => dispatch(addFavorite(result)))
  }
}

export const fetchAndDeleteFavorite = (id, favorite_id) => {
  return (dispatch) => {
    return deleteFavorite(id, favorite_id)
      .then(result => {
        dispatch(removeFavorite(favorite_id))
      })
  }
}


export const removeFavorite = (movieId) => {
  return ({
    type: 'REMOVE_FAVORITE',
    movieId
  })
}

