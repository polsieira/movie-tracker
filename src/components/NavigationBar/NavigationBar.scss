$color: #fff;

.NavigationBar {
  align-items: center;
  background: #1AAB8A;
  display: flex;
  justify-content: space-between;
  min-height: 90px;
  opacity: 1;
  position: fixed;
  width: 100%;
}

.sign-in{
  background:#1AAB8A;
  border: none;
  border-radius: 2px;
  color: #fff;
  cursor: pointer;
  height: 50px;
  font-size: 1em;
  outline: none;
  margin-right: 30px;
  padding: 0 2em;
  position: relative;
  transition: 800ms ease all;
  &:hover{
    background: #fff;
    color: #C3403F;
  }
  &:before,
  &:after {
    background: #1AAB8A;
    content: '';
    height: 2px;
    position: absolute;
    right: 0;
    top: 0;
    transition: 400ms ease all;
    width: 0;
  }
  &:after {
    bottom: 0;
    left: 0;
    right: inherit;
    top: inherit;
  }
  &:hover:before, 
  &:hover:after {
    transition: 800ms ease all;
    width: 100%;
  }
}

.heading {
  color: white;
  font-family: 'Playfair Display', serif;
  left: 45%;
  position: absolute;
  top: 1%;
}

form {
    margin: 30px;
    transform: translate(-0%,-0%);
    transition: all 1s;
    width: 50px;
    height: 50px;
    background: white;
    box-sizing: border-box;
    border-radius: 25px;
    border: 4px solid white;
    padding: 5px;
}

input{
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;;
    height: 42.5px;
    line-height: 30px;
    outline: 0;
    border: 0;
    display: none;
    font-size: 1em;
    border-radius: 20px;
    padding: 0 20px;
}

.fa{
    box-sizing: border-box;
    padding: 10px;
    width: 42.5px;
    height: 42.5px;
    position: absolute;
    top: 0;
    right: 0;
    border-radius: 50%;
    color: #07051a;
    text-align: center;
    font-size: 1.2em;
    transition: all 1s;
}

form:hover{
    width: 200px;
    cursor: pointer;
}

form:hover input{
    display: block;
}

form:hover .fa{
    background: #C3403F;
    color: white;
}