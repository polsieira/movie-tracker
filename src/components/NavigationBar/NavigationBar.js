import React from 'react';
import { Link } from 'react-router-dom';
import './NavigationBar.scss';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';

const NavigationBar = ({ isSignedIn, name, loginUser }) => {

  return (
    <div className='NavigationBar'>
      <form className='nav-form' action="">
        <input className='nav-input' type="search" />
        <i className="fa fa-search"></i>
      </form>
      <h1 className="heading">moooovies.</h1>
      {isSignedIn ? <p className='user'>{name}</p> : null}
      <div className='login-fav'>
        <Link to='/login'>
          {isSignedIn ? <button className='sign-out' type='button' onClick={() => loginUser({ name: '', id: '', isSignedIn: false })}>Sign Out</button> : <button className='sign-in' type='button' >Sign In</button>}
        </Link>
        <Link to='/favorites'>
          <button className='favorites-button' type='button'>Favorites</button>
        </Link>
      </div>
    </div>
  )
}

const mapStateToProps = ({ user }) => ({
  isSignedIn: user.isSignedIn,
  name: user.name
});

const mapDispatchToProps = dispatch => ({
  loginUser: userInfo => dispatch(loginUser(userInfo))
})

export default connect(mapStateToProps, mapDispatchToProps)(NavigationBar);