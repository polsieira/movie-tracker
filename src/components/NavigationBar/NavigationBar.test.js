import { NavigationBar, mapStateToProps, mapDispatchToProps } from './NavigationBar';
import { loginUser } from '../../actions';
import { shallow } from 'enzyme';
import React from 'react';


describe('NavigationBar', () => {
  let wrapper;
  describe('NavigationBar component', () => {
    const mockLoginUser = jest.fn();
    beforeEach(() => {
      wrapper = shallow(<NavigationBar
        isSignedIn={false}
        name='Pol'
        loginUser={mockLoginUser}
      />)
    })

    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

  });

  describe('mapStateToProps', () => {
    it('should return an object with name and isSignedIn', () => {
      // Setup
      const mockState = {
        favorites: [{
          title: 'Wicked',
          release_date: '2017',
          average_rating: 6.5,
          popularity: 55.55,
          movie_id: 5
        },
        {
          title: 'Joker',
          release_date: '2019',
          average_rating: 7.5,
          popularity: 68.55,
          movie_id: 6
        }],
        user: {
          name: 'Pol',
          id: 2,
          isSignedIn: true
        }
      };
      const expected = {
        name: 'Pol',
        isSignedIn: true
      };

      // Execution
      const mappedProps = mapStateToProps(mockState);

      // Expectation
      expect(mappedProps).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', () => {
    it.skip('calls dispatch with an loginUser action when button is clicked', () => {
      // Setup
      const mockDispatch = jest.fn();
      const actionToDispatch = loginUser({ name: '', id: '', isSignedIn: false })

      // Execution
      // const mappedProps = mapDispatchToProps(mockDispatch);
      wrapper.find('button').at(0).simulate('click');

      // Expectaion
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });

})