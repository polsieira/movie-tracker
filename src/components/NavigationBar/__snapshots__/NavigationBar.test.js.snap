// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar NavigationBar component should match a snapshot 1`] = `
<div
  className="NavigationBar"
>
  <form
    action=""
    className="nav-form"
  >
    <input
      className="nav-input"
      type="search"
    />
    <i
      className="fa fa-search"
    />
  </form>
  <h1
    className="heading"
  >
    moooovies.
  </h1>
  <div
    className="login-fav"
  >
    <Link
      to="/login"
    >
      <button
        className="sign-in"
        type="button"
      >
        Sign In
      </button>
    </Link>
    <Link
      to="/favorites"
    >
      <button
        className="favorites-button"
        type="button"
      >
        Favorites
      </button>
    </Link>
  </div>
</div>
`;
